keps/sig-node/2625-cpumanager-policies-thread-placement/README.md

```diff
@@ -0,0 +1,326 @@
+The static policy of the cpumanager already allows to prevent virtual CPU sharing.
+However, for some classes of these latency-sensitive applications running on simultaneous multithreading (SMT) enabled system, it is also beneficial
+to consider thread-level allocation, to avoid physical CPU sharing and prevent possible interferences caused by noisy neighborhoods.
+
+### Goals
+
+* Allow the workload to request the core allocation at hardware-thread level, avoiding noisy neighbours situations
+* Allow the workload to request full physical core allocation, to enable more efficient cache sharing
+
+## Proposal
+
+### User Stories
+
+#### Latency-sensitive applications runtime guarantees
+
+Some classes of latency-sensitive applications (CNF, HFT, ML/AI) benefit of the thread placement constraints this policy enables.
+More precise thread placement allows to control which physical cores are shared among containers, such as:
+
+1. workloads may ensure the physical cores are shared among their threads for increased efficiency
+2. workloads may ensure the physical cores are *not* shared with other containers for interference prevention.
+
+An implementation of the concepts proposed here is already found in [external projects](https://github.com/nokia/CPU-Pooler#hyperthreading-support).
+[OpenStack](https://specs.openstack.org/openstack/nova-specs/specs/mitaka/implemented/virt-driver-cpu-thread-pinning.html),
+which is one of the leading platform for VNF (Virtual Network Functions), the predecessor of CNFs.
+
+#### Improve the density of running containers
+
+Thread allocation guarantees enables more efficient usage of the node resources.
+The nodes can now accommodate safely mixed workloads of latency-sensitive and not-latency-sensitive (infrastructure) pods.
+This increases the node usage, which reduces the need for extra hardware, which drives down the TCO of a container-based solution.
+
+### Risks and Mitigations
+
+This new behaviour is opt-in. Users will need to explicitly enable it in their kubelet configuration. The change is very self contained, with little impact in the shared codebase.
+The impact in the shared codebase will be addressed enhancing the current testsuite.
+
+| Risk                                                      | Impact        | Mitigation |
+| --------------------------------------------------------- | ------------- | ---------- |
+| Bugs in the implementation lead to kubelet crash | High | Disable the policy and restart the kubelet. The workload will run but with weaker guarantees - like it was before this change. |
+
+
+## Design Details
+
+### Proposed Change
+
+We propose to add a new flag in Kubelet called `CPUManagerPolicyOptions` in the kubelet config or command line argument called `cpumanager-policy-options` which allows the user to specify the CPU Manager policy option. If the value of this option is specified to be `smtalign`, it results in further refinements of the existing static policy.
```

Comment:
Very fair point. I moved from the `smtaware` to `smtalign` because I was hoping to keep a short name, but it seems it's too hard, and `smtalign` is indeed misleading. So let's brainstorm some longer names.

What about:
(I will doublecheck later if the kubelet convension is dash-separared-names or CamelCaseNames)
- RejectNonSMTAligned
- ForceSMTAligned
- RequireSMTAligned
- ...